controller/src/main/java/org/jboss/as/controller/extension/ExtensionRegistry.java

```diff
@@ -532,8 +533,12 @@ public SubsystemRegistration registerSubsystem(String name, ModelVersion version
             if (deprecated){
                 ControllerLogger.DEPRECATED_LOGGER.extensionDeprecated(name);
             }
-            return new SubsystemRegistrationImpl(name, version,
-                    profileRegistration, deploymentsRegistration, extensionRegistryType, extension.extensionModuleName, processType);
+            SubsystemRegistrationImpl result =  new SubsystemRegistrationImpl(name, version,
+                                profileRegistration, deploymentsRegistration, extensionRegistryType, extension.extensionModuleName, processType);
+            if (registerTransformers){
+                transformerRegistry.loadAndRegisterTransformers(name, version, extension.extensionModuleName);
```

Comment:
Fixes, there is not extra check to register only transformers that are for same subsystem as subsystem we are registering.